api/request.go

```diff
@@ -3,21 +3,45 @@ package api
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/equinor/vds-slice/internal/cache"
+	"github.com/equinor/vds-slice/internal/vds"
 )
 
 type RequestedResource struct {
 	// The blob url to a vds in form
 	// https://account.blob.core.windows.net/container/blob
 	Vds string `json:"vds" binding:"required" example:"https://account.blob.core.windows.net/container/blob"`
 	// A valid sas-token with read access to the container specified in Vds
-	Sas string `json:"sas,omitempty" binding:"required" example:"sp=r&st=2022-09-12T09:44:17Z&se=2022-09-12T17:44:17Z&spr=https&sv=2021-06-08&sr=c&sig=..."`
+	Sas string `json:"sas,omitempty" example:"sp=r&st=2022-09-12T09:44:17Z&se=2022-09-12T17:44:17Z&spr=https&sv=2021-06-08&sr=c&sig=..."`
 }
 
-type Request interface {
+type Stringable interface {
 	toString() (string, error)
 }
+type Request interface {
```

Comment:
Yeah! Erland and I discussed creating another approach with setters and getter functions, but the problem is that it will make the code line messier. And a lot of repletion of code as all the requests struct must implement those functions. So, we followed this solution which needs minimum change in the existing code. 

Yes, I can the name of the interface to ´IRequestedResource´

Reply:
I agree that `Request` is too general. But wrt `IRequestedResource` I've yet to see any go project using `I` to indicate interface in Go code.  I know using `I` for interfaces is common in java/c# and also C++. But there (at least in C++) we do it because there are no concept of an interface in the language and interfaces are implemented as abstract base classes/structs. I.e. there is no type difference between the "interface" and the concrete class and hence we try to help using the `I` naming convention. However, in Go, interface is its own type, distinct from a struct and thus using `I` in the name doesn't add anything and borders on Hungarian notation [1]. 

What Effective Go [2] says about interface naming:

> Interfaces ..., and are usually given a name derived from the method, such as io.Writer for something that implements Write.

Which makes a lot of sense, as Interfaces in Go provide a way to specify the behavior of an object. Hence the name should reflect the behaviour [2]. 

I suggest renaming `ExtractSas` to something like `NormalizeConnection`, and make the interface `Normalizable` [3]. (also, the `sanitizeSas` should now be inlined in `ExtractSas` imo, as a separate commit ofc)


Btw, my name is spelled Erlend, not Erland 😉 

[1] https://en.wikipedia.org/wiki/Hungarian_notation
[2] https://go.dev/doc/effective_go#interfaces
[3] https://www.oxfordreference.com/display/10.1093/oi/authority.20110803100238574